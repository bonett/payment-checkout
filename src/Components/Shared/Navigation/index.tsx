/* eslint-disable */
import React from "react";
import { Link } from "react-router-dom";
import Logo from 'Assets/Img/logo.png';

function Navigation() {
    return (
        <nav className="nav-content">
            <div className="container">
                <div className="nav-content_wrapper">
                    <Link to={'/'}>
                        <img src={Logo} alt={'pay'} className="nav-content_logo" />
                    </Link>
                    <div>
                        <button className="btn-outlined">
                            Last Purchased
                        </button>
                    </div>
                </div>
            </div>
        </nav>
    );
}

export default Navigation;
