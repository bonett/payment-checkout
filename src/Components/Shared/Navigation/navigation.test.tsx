import React from "react";
import { render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import Navigation from "./";

describe("Navigation component", () => {
  test("Should renders navigation component", () => {
    const { container } = render(
      <BrowserRouter>
        <Navigation />
      </BrowserRouter>
    );

    const image = screen.getByAltText("pay");
    expect(image).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  test("Should validate logo img", () => {
    const { container } = render(
      <BrowserRouter>
        <Navigation />
      </BrowserRouter>
    );

    const image = screen.getByAltText("pay");
    expect(image).toHaveAttribute("src", "logo.png");
    expect(container).toMatchSnapshot();
  });
});
