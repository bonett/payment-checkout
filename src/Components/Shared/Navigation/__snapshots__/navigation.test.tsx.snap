// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation component Should renders navigation component 1`] = `
<div>
  <nav
    class="nav-content"
  >
    <div
      class="container"
    >
      <div
        class="nav-content_wrapper"
      >
        <a
          href="/"
        >
          <img
            alt="pay"
            class="nav-content_logo"
            src="logo.png"
          />
        </a>
      </div>
    </div>
  </nav>
</div>
`;

exports[`Navigation component Should validate logo img 1`] = `
<div>
  <nav
    class="nav-content"
  >
    <div
      class="container"
    >
      <div
        class="nav-content_wrapper"
      >
        <a
          href="/"
        >
          <img
            alt="pay"
            class="nav-content_logo"
            src="logo.png"
          />
        </a>
      </div>
    </div>
  </nav>
</div>
`;
